document how to use :literalref: for code-style external links

ru-fu opened the pull request:
Draft for now until the PR on lxd-sphinx-extensions is merged.

pmatulis commented on .sphinx/requirements.txt:
```diff
@@ -6,6 +6,6 @@ sphinx-tabs
 pyspelling
 sphinxext-opengraph
-lxd-sphinx-extensions
+git+https://github.com/canonical/lxd-sphinx-extensions@refs/pull/26/merge#egg=lxd-sphinx-extensions
```
Shouldn't we be using a doc-specific repo? It looks really odd to be seeing LXD in there. Seems to me we have a doc-tools repo somewhere.

ru-fu replied:
See https://github.com/canonical/lxd-sphinx-extensions/issues/25 - we need to rename the repo, but I want to check with @evildmp about the name first to avoid having to rename it again later.

ru-fu commented on style-guide.rst:
```diff
@@ -146,6 +146,7 @@ For external links, use one of the following methods.
 
 Link inline:
   Define occasional links directly within the surrounding text.
+  To make the link text show up in code-style (which excludes it from the spelling check), use the custom ``:literalref:`` role (requires the :literalref:`lxd-sphinx-extensions <https://github.com/canonical/lxd-sphinx-extensions>` package).
```
How about getting rid of it completely and making it clear that the style guide applies to the starter pack?
There we load all the "custom" extensions, so we don't really need to make a difference.

This might be confusing for people who find this style guide and don't use the starter pack, but I don't think that's our problem. ;-)

ru-fu replied:
My take on this is that we recommend using the starter pack and its corresponding style guide (this one here).
If teams want to do something different, they are free to do so, but they need to maintain their own style guide then.

ru-fu commented:
Updated to get rid of the mention of "custom" extensions - the style guide is meant to be used with the starter pack.
Also removed the requirements change now since the updated extension is now published.
